### DISCUSSION TITLE
Field number mistaken machine name

### DISCUSSION THREAD

**Discussion by Archilegt** [NaturalHistoryMuseum/scratchpads2#6231] [General]
Location: In **Specimen/Observation** content type, vertical tab **Collection**.
Issue: The machine name for the label **Field number** is "field_number" but it should be "field_field_number".
Could you please fix it?

**Comment by PaulKiddle** [comment]
It seems to follow the same format for the other fields, what's the problem it's causing?

**Comment by Archilegt** [comment]
@PaulKiddle, it is not causing a technical problem but an understanding problem.
It does not follow the same format than the other fields. The formatting premises are:
1) Start every machine name with "field_".
2) Add a DarwinCore term as suffix to the machine name.  

Formatting premise 2) is violated because the matching DarwinCore term is not "number" but "field_number". The composite machine name is then "field_field_number".

**Comment by PaulKiddle** [comment]
Thanks, I see.

The worry I have about changing this is that Scratchpads is a very fragile piece of software, and changing things like field/machine names can have unexpected effects that we can't practically test for. I don't think this danger outweighs the benefits.

**Comment by Archilegt** [comment]
Agreed. 
Current solution: Register the exception to the formatting premises for this machine name in Scratchpads documentation and link it to this issue if possible. 
On this [wiki page](https://github.com/NaturalHistoryMuseum/scratchpads2/wiki/Add-locations-and-specimens), the following documentation links are broken:
1) https://github.com/NaturalHistoryMuseum/scratchpads2/wiki/glossary-glossary-darwincore
2) https://github.com/NaturalHistoryMuseum/scratchpads2/wiki/glossary-glossary-bis
3) https://github.com/NaturalHistoryMuseum/scratchpads2/wiki/glossary-glossary-gbif

When the links are fixed, the machine name format exception could be added as a note to 1) or to all three of them. I cannot tell now because I cannot see the content.

Future solution: Open a test site and a branch. I volunteer to populate the site with testing content before the machine name is changed and to evaluate the effects after the machine name is changed. If the dependencies of that field are provided, I will have a more complete idea on what to add and to evaluate.

**Comment by PaulKiddle** [comment]
> Future solution: Open a test site and a branch. I volunteer to populate the site with testing content before the machine name is changed and to evaluate the effects after the machine name is changed. If the dependencies of that field are provided, I will have a more complete idea on what to add and to evaluate.

The problem is, because of the way Drupal's internals work, it's incredibly hard to find references to things like fields. Often they're obscured away by passed variables and string concatenation.
